src/webextension/ticker-timer-modes.js

```diff
@@ -37,27 +37,30 @@ function set_listeners_for_timer_mode(mode) {
     }
 };
 
-async function get_current_url_default() {
-    // returns a promise that resolves to the url of the active window/tab
+async function get_current_url_internal(mode) {
     try {
-        let tabs = await browser.tabs.query({currentWindow: true, active: true});
-        return new URL(tabs[0].url);
-
+        let tabs = await browser.tabs.query({currentWindow: true, active: true}),
+            tab = tabs[0];
+        if (tab.incognito) {
+            // mainly visual confirmation ("we are not logging")
+            // (but also, exception in popup: getBackgroundPage() returns undefined)
+            browser.browserAction.disable(tab.id);
```

Comment:
I don't think we should disable the browserAction button because that is how the user opens the summary page, changes the timer mode, and (depending on settings) sees the total time for the day.  Those things should still be accessible.  Instead we could show a "non-logging" message at the top of the popup panel, like "Private / 0:12".

Reply:
The browserAction popup does not render correctly, and neither does the statistic page, when used from incognito mode. This can be reproduced with the current AMO version too. The button is only disabled for the current tab, it stays enabled in other non-incognito windows.

Showing the time anyway would be a nice feature, but then as a user I woldn't be confident that the logging is actually disabled (no space to display an explanation with the popup disabled).